src/ansys/units/quantity.py

```diff
@@ -299,7 +299,13 @@ def __add__(self, __value):
         if not isinstance(__value, ansunits.Quantity):
             __value = ansunits.Quantity(__value)
         new_units = (self._unit + __value._unit) or self.units
-        new_value = self.value + __value.value
+        if new_units != self.units or new_units != __value.units:
```

Comment:
Please check your logic here. Do you mean `and`? It reads nice and clear if you use `not in`. It's especially good for avoiding bugs.

Reply:
logic updated